styleguide/src/Forms/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+
+const RadioButtonComponent = (
+  {
+    label,
+    id,
+    name,
+    className,
+    onChange,
+    onFocus,
+    onBlur,
+    focus = false,
+    checked = false,
+    disabled,
+    ...props
+  }: RadioButtonProps,
+  ref: React.ForwardedRef<HTMLInputElement>
+): JSX.Element => {
+  const [isChecked, setChecked] = React.useState(checked)
+  const [hasFocus, setFocus] = React.useState(focus)
+
+  const toggleClasses = disabled
+    ? `pointer-events-none border-card-stroke text-on-base-2 font-semibold peer-checked:bg-base-3`
+    : `peer-checked:bg-primary cursor-pointer`
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (disabled) return
+    setChecked(event.target.checked)
+    onChange && onChange(event)
+  }
+  const handleFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+    setFocus(true)
+    onFocus && onFocus(event)
+  }
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    setFocus(false)
+    onBlur && onBlur(event)
+  }
+
+  return (
+    <div
+      className={`radioContainer relative inline-flex justify-center items-center text-f6 text-inverted-1 font-semibold`}
+    >
+      <input
+        className="absolute opacity-0 w-full h-full"
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+      />
+      <label
+        className={`${className} relative inline-flex justify-center items-center border border-base-4 rounded-full w-4 h-4 mr-1 ${
+          hasFocus ? 'border-focus border-2' : ''
```

Comment:
Não precisa ser controlado via JS, da pra fazer apenas com CSS, usando o `focus-within:`
https://tailwindcss.com/docs/hover-focus-and-other-states#focus-within